src/vm/contexts.rs

```diff
@@ -531,7 +531,7 @@ impl EventBatch {
 
 impl<'a> OwnedEnvironment<'a> {
     #[cfg(test)]
-    pub fn new(database: ClarityDatabase<'a>) -> OwnedEnvironment<'a> {
+    pub fn new(database: ClarityDatabase<'a>, use_mainnet: bool) -> OwnedEnvironment<'a> {
         let epoch = StacksEpochId::Epoch2_05;
         OwnedEnvironment {
             context: GlobalContext::new(false, database, LimitedCostTracker::new_free(), epoch),
```

Comment:
I think `false` on line 537 should be updated to `use_mainnet` -- I don't think it's causing your test failures though.

Reply:
I reverted this change. It wasn't necessary to get this round of tests to pass, so I think it's better to keep the PR as small as possible.